core/src/utils.jl

```diff
@@ -899,3 +897,7 @@ function Base.getindex(fv::FlatVector, i::Int)
     v = fv.v[d + 1]
     return v[r + 1]
 end
+
+function preallocation_dispatch(var, value)
+    return isa(var, DiffCache) ? get_tmp(var, value) : var
+end
```

Comment:
That dispatch is indeed nicer. I'm having a hard time coming up with a docstring for this because I do not understand exactly what `get_tmp` does.

Reply:
It looks like instead of `preallocation_dispatch` you can just use `get_tmp` directly. It already has a fallback if the first argument is not a DiffCache: https://github.com/SciML/PreallocationTools.jl/blob/0c41450830881f87daedea946512eb06f956d5f2/src/PreallocationTools.jl#L143